src/components/IDInput.vue

```diff
@@ -90,31 +90,31 @@ export default createComponent({
             if (valid.value) {
                 const parts = state.input.split(".");
                 if (props.file) {
-                    state.file = {
-                        shard: parseInt(parts[0]),
-                        realm: parseInt(parts[1]),
-                        file: parseInt(parts[2])
-                    };
+                    state.file = FileId.fromString(
```

Comment:
You can remove the `parseInt` calls here

Reply:
:+1: